libvcx/src/disclosed_proof.rs

```diff
@@ -273,303 +219,108 @@ pub fn build_rev_states_json(credentials_identifiers: &mut Vec<CredInfo>) -> Vcx
     Ok(rtn.to_string())
 }
 
-impl DisclosedProof {
-    fn create_with_request(source_id: &str, proof_req: &str) -> VcxResult<DisclosedProof> {
-        trace!("create_with_request >>> source_id: {}, proof_req: {}", source_id, proof_req);
-
-        let mut proof: DisclosedProof = Default::default();
-
-        proof.set_source_id(source_id);
-        proof.set_proof_request(proof_req)?;
-        proof.set_state(VcxStateType::VcxStateRequestReceived);
-
-        Ok(proof)
-    }
-
-    fn set_proof_request(&mut self, proof_req: &str) -> VcxResult<()> {
-        let proof_req: ProofRequestMessage = serde_json::from_str(proof_req)
-            .map_err(|err| VcxError::from_msg(VcxErrorKind::InvalidJson, format!("Cannot deserialize proof request: {}", err)))?;
-        self.proof_request = Some(proof_req);
-        Ok(())
-    }
-
-    fn get_state(&self) -> u32 {
-        trace!("DisclosedProof::get_state >>>");
-        self.state as u32
-    }
-    fn set_state(&mut self, state: VcxStateType) {
-        trace!("DisclosedProof::set_state >>> state: {:?}", state);
-        self.state = state
-    }
-
-    fn retrieve_credentials(&self) -> VcxResult<String> {
-        trace!("DisclosedProof::set_state >>>");
+pub fn build_schemas_json(credentials_identifiers: &Vec<CredInfo>) -> VcxResult<String> {
```

Comment:
Seems like this is duplication of `build_credential_defs_json` in`proof_utils.rs`

Reply:
Good catch!